src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -286,10 +286,9 @@ CHIP_ERROR WiFiManager::GetWiFiInfo(WiFiInfo & info) const
 
     if (status.state >= WIFI_STATE_ASSOCIATED)
     {
-        uint8_t mac_string_buf[sizeof("xx:xx:xx:xx:xx:xx")];
         net_sprint_ll_addr_buf(reinterpret_cast<const uint8_t *>(status.bssid), WIFI_MAC_ADDR_LEN,
-                               reinterpret_cast<char *>(mac_string_buf), sizeof(mac_string_buf));
-        info.mBssId        = ByteSpan(mac_string_buf, sizeof(mac_string_buf));
+                               reinterpret_cast<char *>(const_cast<uint8_t *>(mMacStringBuf)), sizeof(mMacStringBuf));
```

Comment:
Wouldn't it be cleaner if `WiFiInfo::mBssId` was a regular array instead of `ByteSpan`? Then you wouldn't need a global state to store BSSID. Also, why do we actually convert the bytes to the hex string? Given how this value is used after #26088, does it even work (the caller of `GetWiFiBssid` uses an array of size 8B)?